File: src/services/adminDashboardService.ts
import api from './api';
import { ApiResponse } from '@/types';
import {
  AdminDashboardBookingsOverview,
  AdminDashboardOverview,
  AdminDashboardRecentBooking,
  AdminDashboardSystemHealth,
  AdminDashboardUserGrowth,
} from '@/types/adminDashboard';

export const adminDashboardService = {
  async getOverview() {
    const response = await api.get<ApiResponse<AdminDashboardOverview>>('/admin/dashboard/overview');
    return response.data;
  },

  async getBookingsOverview(range = '30d') {
    const response = await api.get<ApiResponse<AdminDashboardBookingsOverview>>('/admin/dashboard/bookings-overview', { params: { range } });
    return response.data;
  },

  async getUserGrowth(range = '30d') {
    const response = await api.get<ApiResponse<AdminDashboardUserGrowth>>('/admin/dashboard/user-growth', { params: { range } });
    return response.data;
  },

  async getSystemHealth() {
    const response = await api.get<ApiResponse<AdminDashboardSystemHealth>>('/admin/dashboard/system-health');
    return response.data;
  },

  async getRecentBookings(limit = 5) {
    const response = await api.get<ApiResponse<AdminDashboardRecentBooking[]>>('/admin/dashboard/recent-bookings', { params: { limit } });
    return response.data;
  },
};

File: src/main/java/com/travel/marketplace/modules/dashboard/service/AdminDashboardService.java
package com.travel.marketplace.modules.dashboard.service;

import com.travel.marketplace.modules.dashboard.dto.AdminDashboardBookingsOverviewResponse;
import com.travel.marketplace.modules.dashboard.dto.AdminDashboardOverviewResponse;
import com.travel.marketplace.modules.dashboard.dto.AdminDashboardRecentBookingResponse;
import com.travel.marketplace.modules.dashboard.dto.AdminDashboardSystemHealthResponse;
import com.travel.marketplace.modules.dashboard.dto.AdminDashboardUserGrowthResponse;

import java.util.List;

public interface AdminDashboardService {
    AdminDashboardOverviewResponse getOverview();

    AdminDashboardBookingsOverviewResponse getBookingsOverview(String range);

    AdminDashboardUserGrowthResponse getUserGrowth(String range);

    AdminDashboardSystemHealthResponse getSystemHealth();

    List<AdminDashboardRecentBookingResponse> getRecentBookings(int limit);
}
